qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -56,22 +58,35 @@ def __init__(
             delta: the finite difference step size for the gradient computation. It has a minimum
                 value of 1e-5.
             max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based on the method mentioned.
         """
         validate_min("threshold", threshold, 1e-15)
         validate_min("delta", delta, 1e-5)
 
+        if gradient is None:
+            gradient = Gradient(grad_method="param_shift")
+
         super().__init__(qubit_converter, solver)
 
         self._threshold = threshold
         self._delta = delta
         self._max_iterations = max_iterations
+        self._gradient = gradient
```

Comment:
Just a comment: Terra's `VQE` class already holds a `gradient` object. In the future we could think about using the same one here. However, I am not sure if these can really be intended for the same use cases. Maybe @Cryoris knows a little more about this.